lib/routes/dealstreetasia/home.ts

```diff
@@ -0,0 +1,83 @@
+import { Route } from '@/types';
+// import cache from '@/utils/cache';
+import ofetch from '@/utils/ofetch'; // Unified request library used
+import { load } from 'cheerio'; // An HTML parser with an API similar to jQuery
+// import puppeteer from '@/utils/puppeteer';
+// import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/home',
+    categories: ['traditional-media'],
+    example: '/dealstreetasia/home',
+    // parameters: { section: 'target section' },
+    radar: [
+        {
+            source: ['dealstreetasia.com/'],
+        },
+    ],
+    name: 'Deal Street Asia - Home',
+    maintainers: ['jack2game'],
+    handler,
+    url: 'dealstreetasia.com/',
+};
+
+async function handler() {
+    // const section = ctx.req.param('section');
+    const items = await fetchPage();
+
+    return items;
+}
+
+async function fetchPage() {
+    const baseUrl = 'https://dealstreetasia.com'; // Define base URL
+
+    const response = await ofetch(`${baseUrl}/`);
+    const $ = load(response);
+
+    const jsonData = JSON.parse($('#__NEXT_DATA__').text());
+    // const headingText = jsonData.props.pageProps.sectionData.name;
+
+    const pageProps = jsonData.props.pageProps;
+    const combinedList = [
+        ...pageProps.topStories,
+        ...pageProps.privateEquity,
+        ...pageProps.ventureCapital,
+        ...pageProps.unicorns,
+        ...pageProps.interviews,
+        ...pageProps.deals,
+        ...pageProps.analysis,
+        ...pageProps.ipos,
+        ...pageProps.opinion,
+        ...pageProps.policyAndRegulations,
+        ...pageProps.people,
+        ...pageProps.earningsAndResults,
+        ...pageProps.theLpView,
+        ...pageProps.dvNewsletters,
+    ].map((item) => ({
+        title: item.post_title || 'No Title',
+        link: item.post_url || '',
+        description: item.post_excerpt || '',
+        pubDate: item.post_date ? new Date(item.post_date).toUTCString() : '',
+        category: item.category_link ? item.category_link.replaceAll(/(<([^>]+)>)/gi, '') : '', // Clean HTML if category_link exists
+        image: item.image_url ? item.image_url.replace(/\?.*$/, '') : '', // Remove query parameters if image_url exists
+    }));
+
+    const items1 = jsonData.props.pageProps.reports;
+    const list1 = items1.map((item) => ({
+        title: item.title || 'No Title',
+        link: item.link || '',
+        description: item.excerpt || '',
+        pubDate: item.date ? new Date(item.date).toUTCString() : '',
+        category: item.category_link ? item.category_link.replaceAll(/(<([^>]+)>)/gi, '') : '', // Clean HTML if category_link exists
+        image: item.image_url ? item.image_url.replace(/\?.*$/, '') : '', // Remove query parameters if image_url exists
+    }));
+
+    const list = [...combinedList, ...list1];
```

Comment:
Please put `.reports` inside the spread array.
https://github.com/DIYgod/RSSHub/pull/16909#discussion_r1780054987 is Comment on **lines 40 to 70** not lines 40 to 58

Reply:
Why is this necessary?

`'reports` has different structure vs other parts of the json.

Reply:
You can use [Logical OR `||`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/Logical_OR) to avoid unnecessary spreading and concatenation

Reply:
Ok thank you.